tests/manage/pv_services/test_rbd_pv_encryption.py

```diff
@@ -0,0 +1,213 @@
+    ManageTest,
+    tier1,
+    skipif_ocs_version,
+)
+from ocs_ci.helpers.helpers import (
+    create_unique_resource_name,
+    create_pods,
+)
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.exceptions import (
+    CommandFailed,
+    KMSResourceCleaneupError,
+    ResourceNotFoundError,
+)
+from ocs_ci.utility import kms
+from ocs_ci.ocs.ocp import OCP
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    argnames=["kv_version"],
+    argvalues=[
+        pytest.param("v1", marks=pytest.mark.polarion_id("OCS-2585")),
+        pytest.param("v2", marks=pytest.mark.polarion_id("OCS-2592")),
+    ],
+)
+@skipif_ocs_version("<4.7")
+class TestRbdPvEncryption(ManageTest):
+    """
+    Test to verify RBD PV encryption
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, kv_version):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+        # Initialize Vault
+        self.vault = kms.Vault()
+        self.vault.gather_init_vault_conf()
+        self.vault.update_vault_env_vars()
+
+        # Check if cert secrets already exist, if not create cert resources
+        ocp_obj = OCP(kind="secret", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        try:
+            ocp_obj.get_resource(resource_name="ocs-kms-ca-secret", column="NAME")
+        except CommandFailed:
+            self.vault.create_ocs_vault_cert_resources()
+
+        # Create vault namespace, backend path and policy in vault
+        self.vault_resource_name = create_unique_resource_name("test", "vault")
+        self.vault.vault_create_namespace(namespace=self.vault_resource_name)
+        self.vault.vault_create_backend_path(
+            backend_path=self.vault_resource_name, kv_version=kv_version
+        )
+        self.vault.vault_create_policy(policy_name=self.vault_resource_name)
+
+        ocp_obj = OCP(kind="configmap", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+
+        # If csi-kms-connection-details exists, edit the configmap to add new vault config
+        try:
+            ocp_obj.get_resource(
+                resource_name="csi-kms-connection-details", column="NAME"
+            )
+            self.new_kmsid = self.vault_resource_name
+            vdict = defaults.VAULT_CSI_CONNECTION_CONF
+            for key in vdict.keys():
+                old_key = key
+            vdict[self.new_kmsid] = vdict.pop(old_key)
+            vdict[self.new_kmsid]["VAULT_BACKEND_PATH"] = self.vault_resource_name
+            vdict[self.new_kmsid]["VAULT_NAMESPACE"] = self.vault_resource_name
+            kms.update_csi_kms_vault_connection_details(vdict)
+
+        except CommandFailed:
+            self.new_kmsid = "1-vault"
+            self.vault.create_vault_csi_kms_connection_details()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Cleanup the KMS resources in vault and OCS
+
+        """
+
+        def finalizer():
```

Comment:
you can move the finalizer into the `setup` fixture and avoid having 2 fixtures




```

Reply:
Done